tests/kernel/interrupt/testcase.yaml

```diff
@@ -46,6 +46,8 @@ tests:
       - xtensa
       # TODO: make test work on this arch
       - mips
+      # current trigger_irq implementation of RISC-V architecture does not trigger interrupts
+      - riscv32
```

Comment:
Hi, I've got some questions regarding this change:
1. the discussion started with only the `it8xxx2_evb` platform in mind (which the PR title reflects), but shifted to disabling this test on all riscv32 platforms. I've tested this testcase on the `hifive1` and on `qemu_riscv32`  platforms and it seem to work just fine?
2. this comment suggest that this testcase won't work correctly on both riscv32 and riscv64, but only disables the former. Why?

Reply:
Indeed it works for other platforms, thanks for that advice, probably I misunderstood Dino-Li explanation and it should only affect it8xxx2 board. I'm changing commit to only exclude ITE with proper comment in code.